src/inputs.rs

```diff
@@ -301,7 +302,7 @@ impl PartitionedInput {
                 // Wake up constantly, because we never know when
                 // input will have new data.
                 if caps.is_some() {
-                    activator.activate();
+                    activator.activate_after(cooldown);
```

Comment:
I am surprised too, I did expect that case to slow down. My guess is that running the dataflow on a single machine with a number of processes == number of threads, cpu cores are spending time polling the input and that can lead to a slow down of the processes doing the work instead, but that's just a guess. Anyway the improvement is so small it could be an artifact of the benchmark itself too.

The change to activate immediately if the input just returned an item makes sense though, so I applied it.

As for performance improvements on the work between 0.16 and this PR, you are right, I'm going to add the current `main` to the benchmarks to make sure what we are looking at the right things here.